corehq/motech/auth.py

```diff
@@ -1,10 +1,76 @@
-import requests
 import re
+from typing import Optional
 from urllib.parse import urljoin
+
+from django.utils.translation import gettext_lazy as _
+
+import attr
+import requests
+from oauthlib.oauth2 import LegacyApplicationClient
+from requests import Session
+from requests.auth import AuthBase, HTTPBasicAuth, HTTPDigestAuth
 from requests.exceptions import RequestException
+from requests_oauthlib import OAuth1, OAuth2Session
+
+
+@attr.s(auto_attribs=True, frozen=True, kw_only=True)
+class OAuth1ApiEndpoints:
+    """
+    Endpoints of a particular OAuth1 API
+    """
+    # URL for token to identify HQ. e.g. '/oauth/request_token' (Twitter)
+    request_token_url: str
+    # URL for user to authorize HQ. e.g. '/oauth/authorize'
+    authorization_url: str
+    # URL to fetch access token. e.g. '/oauth/access_token'
+    access_token_url: str
```

Comment:
`auto_attribs=True` is new to me.  Look at this crazy futuristic world we're in.

Reply:
:joy: 

They look a lot like [dataclasses](https://docs.python.org/3/library/dataclasses.html), don't they? :smirk: